x-pack/plugins/canvas/public/components/asset_manager/index.js

```diff
@@ -6,23 +6,61 @@
 
 import { connect } from 'react-redux';
 import { compose, withProps } from 'recompose';
+import { set } from 'lodash';
 import { notify } from '../../lib/notify';
 import { getAssets } from '../../state/selectors/assets';
 import { removeAsset } from '../../state/actions/assets';
+import { elementsRegistry } from '../../lib/elements_registry';
+import { fromExpression, toExpression } from '../../../common/lib/ast';
+import { addElement } from '../../state/actions/elements';
+import { getSelectedPage } from '../../state/selectors/workpad';
 import { AssetManager as Component } from './asset_manager';
 
 const mapStateToProps = state => ({
   assets: Object.values(getAssets(state)), // pull values out of assets object
+  selectedPage: getSelectedPage(state),
 });
 
-const mapDispatchToProps = {
-  removeAsset,
+const mapDispatchToProps = dispatch => ({
+  addImageElement: pageId => asset => {
+    const imageElement = elementsRegistry.get('image');
+    const elementAST = fromExpression(imageElement.expression);
+    const selector = ['chain', '0', 'arguments', 'dataurl'];
+    const subExp = [
+      {
+        type: 'expression',
+        chain: [
+          {
+            type: 'function',
+            function: 'asset',
+            arguments: {
+              _: [asset.id],
```

Comment:
nit: since you're only using the asset's ID here, it would be nice of the handler just took the ID, like `copyAsset` and `removeAsset` does (as `assetId`).